src/traces/bar/plot.js

```diff
@@ -455,17 +622,28 @@ function toMoveInsideBar(x0, x1, y0, y1, textBB, opts) {
     var isHorizontal = !!opts.isHorizontal;
     var constrained = !!opts.constrained;
     var angle = opts.angle || 0;
-    var anchor = opts.anchor || 'end';
+    var anchor = opts.anchor;
```

Comment:
I am not sure if we could safely drop `|| 'end'` here?

Reply:
@archmoj I believe this is safe as I moved the `|| 'end'` to happen earlier, on line 497 (will tag you)

Reply:
Thanks. I saw https://github.com/plotly/plotly.js/pull/6761/files/71b53c3804d40184b83c39b54be7d24d33e8fc26#r1457665683.
Resolving.